fedimintd/src/bin/main.rs

```diff
@@ -71,23 +78,48 @@ async fn main() -> anyhow::Result<()> {
         registry.init();
     }
 
-    let (sender, mut receiver) = tokio::sync::mpsc::channel(1);
-
-    if let Some(ui_port) = opts.ui_port {
-        // Spawn UI, wait for it to finish
-        tokio::spawn(run_ui(opts.cfg_path.clone(), sender, ui_port));
-        receiver
-            .recv()
-            .await
-            .expect("failed to receive setup message");
+    let mut task_group = TaskGroup::new();
+    let (ui_sender, mut ui_receiver) = tokio::sync::mpsc::channel(1);
+
+    // Run admin UI if a socket address was given for it
+    if let Some(ui_bind) = opts.ui_bind {
+        // Make sure password is set
+        let password = match opts.password.clone() {
+            Some(password) => password,
+            None => {
+                eprintln!("fedimintd admin UI requires FM_PASSWORD environment variable to be set");
+                std::process::exit(1);
+            }
+        };
+
+        // Spawn admin UI
+        let cfg_path = opts.cfg_path.clone();
+        let ui_task_group = task_group.make_subgroup().await;
+        task_group
+            .spawn("admin-ui", move |_| async move {
+                run_ui(cfg_path, ui_sender, ui_bind, password, ui_task_group).await;
+            })
+            .await;
+
+        // If federation configs (e.g. local.json) missing, wait for admin UI to report DKG completion
+        let local_cfg_path = opts.cfg_path.join(LOCAL_CONFIG).with_extension(JSON_EXT);
```

Comment:
Here I'm trying to check whether we have enough configs to run bitcoind. This obviously isn't a sufficient check. Is there a better way to accomplish this?